cpp/src/arrow/python/iterators.h

```diff
@@ -27,10 +27,19 @@ namespace arrow {
 namespace py {
 namespace internal {
 
-// Visit the Python sequence, calling the given callable on each element.
-// If the callable returns a non-OK status, iteration stops and the status is returned.
-template <class UnaryFunction>
-inline Status VisitSequence(PyObject* obj, UnaryFunction&& func) {
+// Visit the Python sequence, calling the given callable on each element.  If
+// the callable returns a non-OK status, iteration stops and the status is
+// returned.
+//
+// The call signature for Visitor must be Visit(PyObject*, bool*
+// keep_going). If keep_going is set to false, the iteration terminates
+//
+// The Visitor function  signal that iteration should terminate
```

Comment:
This line seems redundant with the above ("If keep_going is set to false, the iteration terminates").

Reply:
removed stale comment